### DISCUSSION TITLE
Server Timings and Query Plans for pure measures

### DISCUSSION THREAD

**Discussion by erapade** [DaxStudio/DaxStudio#678] [Q&A] [answered]
Hi, Maybe this has already been discussed, but I didn't find anything while searching.

Why don't I get the ServerTiming and QueryPlan results when just evaluating pure measures without iterating over columns?
I.E. in the below examples, only the last one generates Timing and Querying data, although all three generates a result

'EVALUATE {[Measure]} //No Timing and Querying data
EVALUATE ROW("test", [Measure]) //No Timing and Querying data
EVALUATE SUMMARIZECOLUMNS ('Table'[Column], "Name", [Measure])`

**Comment by marcosqlbi** [comment, marked as answer]
It depends on the measure. If you write a measure that uses COUNTROWS ( table ) or DISTINCTCOUNT ( table[column] ) without any filter, the engine doesn't have to perform any calculation and retrieves the result from the metadata.
If you use a measure like SUM ( table[column] ) you should see the execution.

**Reply by erapade** [reply]
Brilliant Marco, Yes that's exactly what I did, a 
`EVALUATE {DISTINCTCOUNT ( table[column] )}`
I also did 
`EVALUATE {CALCULATE( DISTINCTCOUNT ( table[column] ), ALL())}`
Yes, the last one is kind of useless since I don't have an outer filter scope, just me trying to understand DAX, FE, and SE. The the last one generated the smallest Query plan I ever seen :-)

**Reply by marcosqlbi** [reply]
Good - so it just works as expected! :)
